be1-go/cli/witness/witness.go

```diff
@@ -42,36 +44,59 @@ func Serve(context *cli.Context) error {
 
 	h := hub.NewWitnessHub(point)
 
-	ws, err := connectToOrganizer(organizerAddr, organizerPort)
+	err = connectToSocket(hub.OrganizerSocketType, organizerAddress, h, organizerPort)
 	if err != nil {
 		return xerrors.Errorf("failed to connect to organizer: %v", err)
 	}
 
-	organizerSocket := hub.NewOrganizerSocket(h, ws)
+	for _, otherWit := range otherWitness {
+		err = connectToSocket(hub.WitnessSocketType, otherWit, h, organizerPort)
+		if err != nil {
+			return xerrors.Errorf("failed to connect to witness: %v", err)
+		}
+	}
 
-	go organizerSocket.WritePump()
-	go organizerSocket.ReadPump()
+	go hub.CreateAndServeWs(hub.WitnessHubType, hub.ClientSocketType, h, clientPort)
+	go hub.CreateAndServeWs(hub.WitnessHubType, hub.WitnessSocketType, h, witnessPort)
 
 	done := make(chan struct{})
-	go h.Start(done)
-
-	hub.CreateAndServeWs(hub.WitnessHubType, hub.ClientSocketType, h, clientPort)
+	h.Start(done)
```

Comment:
Maybe I'm missing something but I reckon `h.Start()` runs infinitely so this would never unblock, making lines 65-67 unreachable

Reply:
I have implemented a solution for this, could you have a look and tell me what you think?